### PR TITLE
Make custom_templates relative to config file

### LINKED ISSUE #477: Make custom_templates relative to mkdocs.yml
**Is your feature request related to a problem? Please describe.**
See https://github.com/mkdocstrings/mkdocstrings/pull/474

Currently the custom templates value is inserted into the templates paths list as is, so it's relative to the current working directory, which doesn't work with the `-f` option of MkDocs.

### PR DESCRIPTION
**Pull request by waylan**
Fixes #477.

I have not even looked at tests, etc. But this should be working code.

### CODE REVIEW

**Comment by waylan** [comment]
I had very limited time today. I will come back to this later. Of you can run with it if you want.

**Comment by waylan** [comment]
It appears that the tests are all failing due to an unrelated issue:

```
 DeprecationWarning: 'materialx.emoji.twemoji' is deprecated.
  Material emoji logic has been officially moved into mkdocs-material
  version 9.4. Please use Material's 'material.extensions.emoji.twemoji'
  as mkdocs_material_extensions is deprecated and will no longer be
  supported moving forward. This is the last release.
```

**Comment by waylan** [comment]
Now that I have added tests, this should be ready to go. I'm ignoring the unrelated failing tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,14 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
 <!-- insertion marker -->
+
+### Bug Fixes
+
+- Make `custom_templates` relative to MkDocs configuration file.
+
```
Comment (pawamoy) [inline CHANGELOG.md:14]: Our changelog is auto-generated, can you revert this?

**Comment by pawamoy** [comment]
Thanks for the tests. It seems the other ones are failing because `pymdownx` is missing from the test group. I'll fix that in another commit/PR :slightly_smiling_face:

File: CHANGELOG.md (hunk 1 above)
Comment (waylan) [reply CHANGELOG.md]: Done

**Pull request merged**